core/services/keeper/registry_interface.go

```diff
@@ -119,8 +121,21 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 		}
 		return activeUpkeeps, nil
 	case RegistryVersion_1_2:
-		// TODO (sc-37024): Get active upkeep IDs from contract in batches
-		return rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(0), big.NewInt(0))
+		activeUpkeepIDs := make([]*big.Int, 0)
+		for {
```

Comment:
another approach you can take in order to not make contract changes is first fetch the numUpkeeps from getState() function and the run iterations based on it

Reply:
Done